src/GraphQL/Extensions/SchemaExtensions.cs

```diff
@@ -121,7 +135,10 @@ public static void AutoRegister(this ISchema schema, Type clrType, AutoRegisteri
         /// <param name="mode">Which registering mode to use - input only, output only or both.</param>
         public static void AutoRegister<TClrType>(this ISchema schema, AutoRegisteringMode mode = AutoRegisteringMode.Both)
         {
-            schema.AutoRegister(typeof(TClrType), mode);
+            if (mode.HasFlag(AutoRegisteringMode.Output))
+                schema.RegisterTypeMapping<TClrType, AutoRegisteringObjectGraphType<TClrType>>();
+            if (mode.HasFlag(AutoRegisteringMode.Input))
+                schema.RegisterTypeMapping<TClrType, AutoRegisteringInputObjectGraphType<TClrType>>();
```

Comment:
this change allows the AOT trimmer to follow the dynamic building of the `AutoRegisteringObjectGraphType<TClrType>` and then other hints prevent trimming of that class

Reply:
Why not mark `Type clrType` parameter from `AutoRegister` with `[DynamicallyAccessedMembers]` ?

Reply:
`AutoRegisteringObjectGraphType<T>` contains the `[DynamicallyAccessedMembers]` attribute on `T`.

I did not place attributes where I did not think it necessary.  For example, most of the `IGraphQLBuilder` extension methods do not have attributes, because they simply call other methods which do have the attribute.

It of course is possible that the attribute is necessary here, perhaps even because of a bug in the compiler.  When we have type-first AOT compiled tests in place then we will know for sure.

Reply:
I think this is good that I have been reading documentation for the past few days. Look - 
![изображение](https://user-images.githubusercontent.com/21261007/212782048-59c139b1-9bb7-4ac7-b51d-69775e77f735.png) from https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/fixing-warnings

Also this maybe the reason of some strage behaviour you have see before. Just add suggested properties into directory.build.targets to see the whole warnings chain.

Reply:
Yeah, I think that's more referring to `Type type` parameters rather than generic parameters (based on the warnings anyway).  Either way, this needs much more work to review all of the build warnings.  Not something I'm about to do just now.

Reply:
So? Could it be fixed for now adding transitive attribute to `Autoregister` instead of copying (inlining) the code?

Reply:
I'm sorry I don't know what you are suggesting.

Reply:
I suggested to mark `Type clrType` parameter from `AutoRegister` with [DynamicallyAccessedMembers] instead of copying code from called methods. Also for `TClrType` generic argument.

Reply:
As you did with `FieldBuilder<TSourceType, TReturnType> Argument` method overloads. As I see these cases are identical.

Reply:
This does not exactly apply here.  The CLR type and its methods do not need to be 'preserved' -- at least, not by this method.  Rather, the graph types based on the CLR type need to be preserved, such as `AutoRegisteringObjectGraphType<TClrType>`.  

This code was changed to use generics in order that the compiler could have static references to the auto registering type and the CLR types used downstream, if you follow me, which will then preserve the members of the CLR type via the attribute marked on the auto registering type.

In the case of the field builders, it is the graph types that are marked, not the CLR types, as the method's resolve function provides the static reference to the compiler.

I'm pretty sure the change as I have it here is correct and required.

Reply:
> I suggested to mark Type clrType parameter from AutoRegister with `[DynamicallyAccessedMembers]`

Perhaps I misunderstood you.  It is probably a good idea to make this change, but I would not change markings on `TClrType`.  It would direct people to use the proper overload, which should work.

Reply:
I added a RequiresUnreferencedCode to a couple methods.

Reply:
ok